### DISCUSSION TITLE
Aftermarket Switch Controllers not really working

### DISCUSSION THREAD

**Discussion by MRKane05** [darthcloud/BlueRetro#307] [Q&A] [answered]
Hello all! I got put onto the BlueRetro project when discussing wireless controllers for a Playstation 2 portable project over at BitBuilt. In my tests it's been nothing short of perfect, and I settled on getting some aftermarket switch controllers with charging ports to make my life easier in the final portable.

Thing is, while I can get them to connect I just can't seem to get proper input out of them. Occasionally a button press works but in general they're kind of dead in the water. When trying to pair them to the ESP32 it seems that the left one always pairs as controller 1, and the right as controller 2. I don't have a switch to test them with at this stage, so I figured I'd ask if anyone else has any insightful knowledge they could share here!

This is the stylish controller here. It's not make or break if I can't get it working, but I really like how light they are and how they sit in the hand :)
https://www.aliexpress.com/item/1005003531502522.html?spm=a2g0o.order_list.0.0.24bc1802cS94F0

**Comment by darthcloud** [comment]
Added to my amazon list:
https://www.amazon.ca/hz/wishlist/ls/2RMFK6DHX29NQ?ref_=wl_share

If someone buy me one I can investigate to add support.

**Reply by MRKane05** [reply]
Is there any way I could somehow use the one I've got to fish for the information you're after? Buying another one is just a mite beyond my price range sadly.

**Reply by darthcloud** [reply]
It's too much trouble doing that remotely.

**Reply by MRKane05** [reply]
I figured that could be the case. I'll have to look at finances and see if I can make something work. Would an alternative of buying you a cheaper one form Aliexpress be an option?

**Reply by MRKane05** [reply]
> Added to my amazon list: https://www.amazon.ca/hz/wishlist/ls/2RMFK6DHX29NQ?ref_=wl_share
> 
> If someone buy me one I can investigate to add support.

Hey dude, got paid so I've made the purchase and hopefully the controller will be winging its way through to you! Thank you so much for the help already and fingers crossed this doesn't turn into too much of a hassle!

Shout at me if you want receipts, tracking information, or that sort of stuff :)

-Kano

**Reply by darthcloud** [reply]
Thanks!, I haven't got any notification from Amazon but it's not in the list anymore, So I guess it show up sometimes soon. Thanks again!

**Reply by darthcloud** [reply]
Received today, thanks!!

Will work on it likely next week.

**Reply by MRKane05** [reply]
That was crazy quick delivery! Thanks very much, and I hope it doesn't turn into too much of a task!

**Reply by MRKane05** [reply]
Did having the controller prove helpful? I've been playing with it in my spare time and wondering if it'll be too cheaply made for you to find useful!

**Reply by darthcloud** [reply]
Yes I haven't have time to get to it. Once v1.7 is complete I will support for a few controller.

Sorry for delay.

**Comment by darthcloud** [comment, marked as answer]
Sorry for delay, finally had time to look at your controller.

It's actually two joycon and so for BlueRetro it's handled as two separate controllers.

They work fine as-is for me but by default will be in the sideway orientation.

To use them a a single controller I added two presets.
Go to https://blueretro.io/blueretro_presets.html and connect to your BlueRetro.

First select output 1 and select "Switch Left Joycon Upright" click save, wait for green text to appear.
<img width="281" alt="chrome_CosPbgOsVu" src="https://user-images.githubusercontent.com/3744056/180513746-d27d6cda-9b59-4c58-8875-885ec6b73f68.png">

Then select output 2 and select "Switch Right Joycon Upright" click save, wait for green text to appear.
<img width="295" alt="chrome_lYNgsZs9ih" src="https://user-images.githubusercontent.com/3744056/180513770-75222de2-fb66-4cef-8e2a-9237ab5b020d.png">

Then pair the joycon with blueretro separately using the sync button hidden on the sliding side.

It's important to connect the left joycon first, then connect the right joycon

After that you can clip them on the middle support and they should function as one controller.

**Reply by MRKane05** [reply]
Sorry about the exceedingly long time required to get back to you here - there's been a lot of water under the bridge. This fixed the problem beautifully! Thank you so much for your support here - this is an amazing project!
